### PR TITLE
docs: 2/3 enable Sphinx nitpicky [pytorch] part 1/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Reaction to #18068 

> [nitpicky](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-nitpicky)
If true, Sphinx will warn about all references where the target cannot be found. Default is False. You can activate this mode temporarily using the [-n](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-n) command-line switch.

ref: https://github.com/optuna/optuna/issues/4213

As part of the changes we have fixed several broken references and not rendered typings in docs

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18484.org.readthedocs.build/en/18484/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Comment by Borda** [comment]
starting from 381 issues and addressing one by one...

File: docs/source-pytorch/accelerators/accelerator_prepare.rst [hunk 1]
```diff
@@ -50,7 +50,7 @@ This will make your code scale to any arbitrary number of GPUs or TPUs with Ligh
         z = torch.Tensor(2, 3)
         z = z.to(x)
 
-The :class:`~lightning.pytorch.core.module.LightningModule` knows what device it is on. You can access the reference via ``self.device``.
```
Comment (awaelchli) [inline docs/source-pytorch/accelerators/accelerator_prepare.rst:53]: Thanks for doing this! These changes fix a lot of useful links.

File: docs/source-pytorch/conf.py [hunk 2]
```diff
@@ -325,6 +325,13 @@ def _load_py_module(name: str, location: str) -> ModuleType:
     "graphcore": ("https://docs.graphcore.ai/en/latest/", None),
     "habana": ("https://lightning-ai.github.io/lightning-Habana/", None),
 }
+nitpicky = True
```
Comment (awaelchli) [inline docs/source-pytorch/conf.py:328]: I would land the current changes and disable this for now until we have resolved all of them :)

**Review by awaelchli — APPROVED** [review]
LGTM. I would land the current improvement in this PR with nitpicky disabled and address the other things separately.

File: docs/source-pytorch/conf.py (hunk 2 above)
Comment (Borda) [reply docs/source-pytorch/conf.py:328]:
```suggestion
nitpicky = False  # TODO: to be continued
```

**Pull request merged**